codec/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java

```diff
@@ -134,6 +134,8 @@ public ByteBuf cumulate(ByteBufAllocator alloc, ByteBuf cumulation, ByteBuf in)
     private boolean singleDecode;
     private boolean decodeWasNull;
     private boolean first;
+    private boolean callingChildDecode;
+    private boolean handlerRemovedPending;
```

Comment:
was not about memory but more about "logic" as it can only be in one state of both. But this is just a nit. So feel free to ignore